ta/pkcs11/src/processing.c

```diff
@@ -990,18 +990,30 @@ enum pkcs11_rc entry_processing_key(struct pkcs11_client *client,
 			goto out;
 
 	} else if (processing_is_tee_asymm(proc_params->id)) {
-		assert(function == PKCS11_FUNCTION_DERIVE);
 
-		rc = init_asymm_operation(session, function, proc_params,
+		rc = init_asymm_operation(session, operation, proc_params,
```

Comment:
Before this change, `init_asymm_operation()` was called with `PKCS11_FUNCTION_DERIVE` for derivation. Now it is called with `PKCS11_FUNCTION_ENCRYPT` for key derivation. I think this will hurt existing implementation:
https://github.com/OP-TEE/optee_os/blob/3.19.0/ta/pkcs11/src/processing_asymm.c#L223
https://github.com/OP-TEE/optee_os/blob/3.19.0/ta/pkcs11/src/processing_asymm.c#L289
https://github.com/OP-TEE/optee_os/blob/3.19.0/ta/pkcs11/src/processing_asymm.c#L172
Could you cross check that?

Reply:
Hi @etienne-lms
Thanks for pointing this out,
init_asymm_operation was moved in switch statement,

https://github.com/OP-TEE/optee_os/pull/5647/commits/00978ca55be13c1da61ec28972c133dc1d67a752